packages/block-editor/package.json

```diff
@@ -31,6 +31,7 @@
 		"@babel/runtime": "^7.13.10",
 		"@wordpress/a11y": "file:../a11y",
 		"@wordpress/blob": "file:../blob",
+		"@wordpress/block-serialization-default-parser": "file:../block-serialization-default-parser",
```

Comment:
Had to add this to pass commit verification. This definitely shouldn't be here.

Should we export this in `@wordpress/blocks`? @gziolo @youknowriad

Reply:
I'm don't think we should directly depend on this package. It would be better to export this from `blocks`, the same way as we export the `parse` from `blocks`.

Reply:
> I don't think we should directly depend on this package. It would be better to export this from blocks, the same way as we export the parse from blocks.

No, these are separate things from separate modules, so I agree with Grzegorz.

`blocks` corresponds to the complete Blocks API, the one that provides blocks with all of their attributes, with automatic validation and migration, etc. Basically, _stage II_, per our previous discussion.

`block-serialization-default-parser` is a parser for the HTML-based block format, so _stage I._ It's not the only parser; in fact, it's not Gutenberg's original parser. The original one is based on the actual Gutenberg grammar specification and is known as the [spec parser](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-serialization-spec-parser/README.md). But PEG parsers are slow, so @dmsnell handcrafted  `block-serialization-default-parser` in both JS and PHP so that we would have a fast implementation by default. There is even a [Rust-based parser](https://github.com/Hywan/gutenberg-parser-rs) that compiles to WebAssembly and other native targets written by former contributor Ivan!

The important thing is that these parsers are all independent of block semantics, i.e. they are distinct from block-specific knowledge such as attribute schemas. They convert HTML into a tree of raw block nodes. And these parsers should all be interchangeable so that developers have the freedom to choose the parsers that best suit their needs. So yes, if `block-editor` now wants to quickly parse HTML into raw blocks, it should list `block-serialization-default-parser` as an explicit dependency. That's the goal of these small packages. :)

Reply:
I concur with that what @gziolo and @mcsf said. stage I and stage II are confusing - maybe it's time to get back #15674

Reply:
Thanks for the explanation, it makes sense! We will leave this as-is then!